cdap-api/src/main/java/co/cask/cdap/api/workflow/WorkflowToken.java

```diff
@@ -71,12 +114,51 @@
   List<NodeValueEntry> getAll(String key);
 
   /**
-   * Get the {@link Map} of key-values that were added to the {@link WorkflowToken}
-   * by specific node.
+   * Same key can be added to the WorkflowToken by multiple nodes.
+   * This method returns the {@link List} of {@link NodeValueEntry}, where
+   * each entry represents the unique node name and the {@link Value} that it set
+   * for the specified key for a given scope.
+   * <p>
+   * The list maintains the order in which the values were
+   * inserted in the WorkflowToken for a specific key except in the case of fork
+   * and join. In case of fork in the Workflow, copies of the WorkflowToken are made
+   * and passed along each branch. At the join, all copies of the
+   * WorkflowToken are merged together. While merging, the order in which the values were
+   * inserted for a specific key is guaranteed within the same branch, but not across
+   * different branches.
+   * @param key the key to be searched
+   * @param scope the {@link WorkflowToken.Scope} for the key
+   * @return the list of {@link NodeValueEntry} from node name to the value that node
+   * added for the input key for a given scope
+   */
+  List<NodeValueEntry> getAll(String key, Scope scope);
+
+  /**
+   * Get the {@link Map} of key to {@link Value}s that were added to the {@link WorkflowToken}
+   * by specific node for a {@link Scope#USER} scope.
    * @param nodeName the unique name of the node
-   * @return the map of key-values that were added by the specified node
+   * @return the map of key to values that were added by the specified node
    */
-  Map<String, String> getAllFromNode(String nodeName);
+  Map<String, Value> getAllFromNode(String nodeName);
+
+  /**
+   * Get the {@link Map} of key to {@link Value}s that were added to the {@link WorkflowToken}
+   * by specific node for a given scope.
+   * @param nodeName the unique name of the node
+   * @param scope the {@link WorkflowToken.Scope} for the key
+   * @return the map of key to values that were added by the specified node for a given scope
+   */
+  Map<String, Value> getAllFromNode(String nodeName, Scope scope);
+
+  /**
+   * Same key can be added to the WorkflowToken by multiple nodes.
+   * This method returns the key to {@link List} of {@link NodeValueEntry}
+   * added in the {@link WorkflowToken.Scope} provided.
+   * @param scope the scope for the key
+   * @return the {@link Map} of key to {@link List} of {@link NodeValueEntry} added for
+   * the given scope
+   */
+  Map<String, List<NodeValueEntry>> getAll(Scope scope);
```

Comment:
To be consistent, there should be `getAll()` as well to return a map for all user scope tokens.